src/main.ts

```diff
@@ -21,14 +21,14 @@ const run = async (): Promise<void> => {
     const {results, summary} = await synthetics.executeTests(reporter, config)
     const resultSummary = renderResults({config, reporter, results, startTime, summary})
     if (
-      resultSummary.criticalErrors > 0 ||
+      (resultSummary.criticalErrors > 0 && config.failOnCriticalErrors) ||
```

Comment:
`resultSummary.criticalErrors` is only [set by unhealthy tests when `config.failOnCriticalErrors` if disabled](https://github.com/DataDog/datadog-ci/blob/5c22d31e4fb6167db93432ebaa4d9dc72dc5c493/src/commands/synthetics/utils.ts#L767-L769).
It serves to [ignore triggering / polling errors thrown by `executeTests`](https://github.com/DataDog/datadog-ci/blob/5c22d31e4fb6167db93432ebaa4d9dc72dc5c493/src/commands/synthetics/command.ts#L95-L106)

Reply:
Hmm ok, I think I'll revert this part of the changes to only address the timeout cases. We can handle this case on another PR (along with a better error handling to align datadog-ci and integrations).